tests/test_skyWcs.py

```diff
@@ -988,45 +988,35 @@ def setUp(self):
                     crpix=crpix,
                     crval=crval,
                     cdMatrix=cd))
-
-    def points(self):
-        for x in (0.0, -2.0, 42.5, 1042.3):
-            for y in (27.6, -0.1, 0.0, 196.0):
-                yield Point2D(x, y)
+        self.pixelPoints = [Point2D(x, y) for x, y in
+                            itertools.product((0.0, -2.0, 42.5, 1042.3),
+                                              (27.6, -0.1, 0.0, 196.0))]
 
     def testGenericWcs(self):
         """Test that input and output points represent the same sky position.
 
         Would prefer a black-box test, but don't have the numbers for it.
         """
+        inPoints = self.pixelPoints
         for wcs1 in self.wcsList:
             for wcs2 in self.wcsList:
                 transform = makeWcsPairTransform(wcs1, wcs2)
-                # check that the transform has been simplified
-                frameSet = transform.getFrameSet()
-                self.assertEqual(frameSet.nFrame, 2)
-                for point1 in self.points():
-                    point2 = transform.applyForward(point1)
-                    self.assertPairsAlmostEqual(
-                        transform.applyInverse(point2),
-                        point1)
-                    self.assertSpherePointsAlmostEqual(
-                        wcs1.pixelToSky(point1),
-                        wcs2.pixelToSky(point2))
+                outPoints = transform.applyForward(inPoints)
+                inPointsRoundTrip = transform.applyInverse(outPoints)
+                self.assertPairListsAlmostEqual(inPoints, inPointsRoundTrip)
+                self.assertSpherePointListsAlmostEqual(wcs1.pixelToSky(inPoints),
+                                                       wcs2.pixelToSky(outPoints))
 
     def testSameWcs(self):
         """Confirm that pairing two identical Wcs gives an identity transform.
         """
         for wcs in self.wcsList:
             transform = makeWcsPairTransform(wcs, wcs)
             # check that the transform has been simplified
```

Comment:
Is this comment still valid? It looks to me that all that's being tested is round-trippinmg, not simplification.

Reply:
Agreed. I added such a test